subsys/bluetooth/host/hci_core.c

```diff
@@ -3387,6 +3398,27 @@ static int le_init(void)
 		net_buf_unref(rsp);
 	}
 
+#if defined(CONFIG_BT_BROADCASTER)
+	if (IS_ENABLED(CONFIG_BT_BROADCASTER)) {
+		/* Read LE Max Adv Data Len */
+		err = bt_hci_cmd_send_sync(BT_HCI_OP_LE_READ_MAX_ADV_DATA_LEN, NULL, &rsp);
+		if (err > 0) {
+			le_read_maximum_adv_data_len_complete(rsp);
+			net_buf_unref(rsp);
+		} else if (err == -EIO) {
+			/* if the command doesn't exist, flag it with 0xffff so
+			 * the host knows that it doesn't know the maximum
+			 * advertising size of the controller
+			 * we can use 0xffff because the maximum length of
+			 * advertising data is 1650
+			 */
+			bt_dev.le.max_adv_data_len = 0xffff;
```

Comment:
If we want to expose this value later to application then maybe we should create a `#define` for it. 

The command is mandatory to support when extended advertising is supported, so if `CONFIG_BT_EXT_ADV=y` then this should be considered an error. 

If the command does not exist and `CONFIG_BT_EXT_ADV=n` then the appropriate value would be `31`, indicating only supporting legacy advertising length.

Reply:
If this is considered an error for ext adv and we know the value for legacy then we don't need any define, right?